File: modules/react-native-sample/android/src/main/java/com/sample/entity/QuoteBaseStatus.kt
package com.sample.entity

enum class QuoteBaseStatus(val type: String) {
    UP("UP"),
    DOWN("DOWN"),
    UNKNOWN("UNKNOWN");

    companion object {
        fun fromString(type: String): QuoteBaseStatus {
            return when (type) {
                "UP" -> UP
                "DOWN" -> DOWN
                else -> UNKNOWN
            }
        }
    }
}

File: modules/react-native-sample/ios/Entity/QuoteBaseStatus.swift
import Foundation

enum QuoteBaseStatus: String {
    case UP = "UP"
    case DOWN = "DOWN"
    case UNKNOWN = "UNKNOWN"
    
    static func fromString(value: String) -> Self{
        switch(value) {
        case "UP":
            return .UP
        case "DOWN":
            return .DOWN
        default:
            return .UNKNOWN
        }
    }
}
